waku/waku_sync/storage.nim

```diff
@@ -0,0 +1,59 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import std/[times, tables], chronicles, chronos, stew/results
+
+import ./raw_bindings
+
+logScope:
+  topics = "waku sync"
+
+type WakuSyncStorageManager* = object
+  storages: OrderedTable[string, Storage]  # Map of dateTime and Storage objects. DateTime is of the format YYYYMMDDHH
+  maxHours: times.Duration
+
+proc new*(T: type WakuSyncStorageManager, hoursToStore: times.Duration = initDuration(minutes=120)): T =
+  return WakuSyncStorageManager(maxHours:hoursToStore.inHours)
+
+#Time should be in YYYYMMDDHH format.
+proc createStorage(
+    self: WakuSyncStorageManager, time: string
+): Result[Storage, string] =
+  let storage:Storage = Storage.new().valueOr:
+    error "storage creation failed"
+    return err("storage creation failed")
+  self.storages[time] = storage
+  return storage
```

Comment:
we need to check if the new overlap with an older one.

Reply:
I kind of did that in the caller of this ` proc retrieveStorage*( `and hence this is not made public. Was trying to modularize too much i think and maybe this proc is not even required. Will refactor.